pkg/controller/egressgateway/egressgateway_controller.go

```diff
@@ -397,7 +398,7 @@ func (r *ReconcileEgressGateway) reconcileEgressGateway(ctx context.Context, egw
 		}
 	}
 
-	openshift := r.provider == operatorv1.ProviderOpenShift
+	openshift := r.provider.IsOpenShift()
```

Comment:
nit: it does not seem necessary to store this in a variable.

Reply:
The `openshift` variable is removed.